great_expectations/cli/datasource.py

```diff
@@ -204,7 +212,11 @@ def _datasource_new_flow(
         selected_database = _prompt_user_for_database_backend()
         helper = _get_sql_yaml_helper_class(selected_database, datasource_name)
     else:
-        helper = None
+        helper = None  # type: ignore[assignment] # causes error below
+        raise AttributeError(
+            "No `helper` selected"
+        )  # `None` has no `send_backend_choice_usage_message()`.
+        # Explicitly raised so that mypy understands that helper is not `None` below this line
```

Comment:
Seems like we might want a cleaner exit here. But this keeps the existing runtime behavior.